applications/StructuralMechanicsApplication/custom_utilities/large_displacement_sensitivity_variables.h

```diff
@@ -0,0 +1,161 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    
+//
+
+#if !defined(KRATOS_LARGE_DISPLACEMENT_SENSITIVITY_VARIABLES_H_INCLUDED )
+#define  KRATOS_LARGE_DISPLACEMENT_SENSITIVITY_VARIABLES_H_INCLUDED
+
+// System includes
+
+// External include
+
+// Project includes
+#include "includes/define.h"
+#include "includes/node.h"
+#include "geometries/geometry.h"
+#include "utilities/matrix_vector_utilities.h"
+#include "custom_utilities/large_displacement_variables.h"
+#include "utilities/geometrical_sensitivity_utility.h"
+
+namespace Kratos
+{
+
+/**
+ * @class LargeDisplacementDifferentialSensitivityVariables
+ * @ingroup StructuralMechanicsApplication
+ * @brief Analytical shape sensitivities for large displacement differential variables.
+ * @details @see LargeDisplacementDifferentialVariables.
+ */
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) LargeDisplacementDifferentialSensitivityVariables
+{
+    public:
+        typedef Geometry<Node<3>> GeometryType;
+
+        explicit LargeDisplacementDifferentialSensitivityVariables(GeometryType const& rGeom);
+
+        double DetJ0(std::size_t IntegrationIndex, ShapeParameter Deriv);
+
+        const Matrix& DN_DX0(std::size_t IntegrationIndex, ShapeParameter Deriv);
+
+        /// Deformation gradient sensitivity.
+        const Matrix& F(std::size_t IntegrationIndex,
+                        ShapeParameter Deriv,
+                        bool IsAxisymmetric = false);
+
+    private:
+        GeometryType const& mrGeom;
+        Matrix mJ0;
+        Matrix mDN_DX0_Deriv;
+        Matrix mF_Deriv;
+        double mDetJ0_Deriv;
+        std::size_t mCurrentIntegrationIndex = -1;
+        ShapeParameter mCurrentShapeDerivative;
+
+        void Synchronize(std::size_t IntegrationIndex, ShapeParameter Deriv);
```

Comment:
a small description of what is done by this function and ideally by its parameters would be very welcome...i don't have a clue of what it does... nor of what ShapeParameter does (and this is used everywhere)